tests/scans/code_analysis/README.md

```diff
@@ -1,63 +1,55 @@
+- Run the test.
+- If the test passes without failures, everything is correct, and no action is needed.
+- If the test fails, `wazuh-qa/tests/scans/code_analysis/new_flaws_{module}.json` file will report the new code vulnerabilities found.
+You should analyze the new vulnerabilities found in the module and report them in GitHub issues.
 
-- `test_python_flaws.py`: the test itself. This test will be passed using the same Python virtual environment used in the Wazuh framework and API unittests.
-If the test fails, a new JSON file will be created in `wazuh-qa/tests/scans/code_analysis` showing information about the possible new vulnerabilities found.
+If you need to fix a new vulnerability, add it to the **to_fix** key module's JSON file entry found in the **known_flaws** directory. 
+If the new vulnerability is a false positive, add it to the **false_positives** key module's JSON file entry found in the **known_flaws** directory. 
 
-### Usage
+The test updates the files inside **known_flaws** automatically with information like the line number or range of the flaws in the **to_fix** dictionary. The test also removes flaws from the **known_flaws** files if Bandit did not report them.
 
-The workaround for this test will be the following:
+## Parameters
 
-- Pass the test.
+You can set the directories, repository, and branch parameters to test any directory containing Python code inside the Wazuh organization.
+You can also use more parameters to customize the test functionality. The test will only succeed if you check different directories and repositories, as we don't have **known_flaws** files for non-default directories.
 
-- If the test passes, no actions are needed, everything is correct.
+> By default, the test checks the `framework`, `wodles` and `api` directories in the [wazuh/wazuh](https://github.com/wazuh/wazuh) repository's master branch.
 
-- If the test fails, new code vulnerabilities will be found in `wazuh-qa/tests/scans/code_analysis/new_flaws_{module}.json`.
-  - We analyze the new vulnerabilities found in the module and report them in GitHub issues.
-  - We move the vulnerabilities to the `to_fix` key of the known flaws JSON file.
-  - If the new vulnerability is considered a false positive, we add it to the `false_positives` list of the dictionary in its respective `known_flaws` JSON file.
-  - If the new vulnerability is a real vulnerability, we solve the problem reported and remove the flaw from the known flaws file.
+- `--repo`: the repository used. 
 
-The test also updates the known_flaws files automatically. If we have a look at a known_flaws file, we will see that each flaw dictionary contains information like the line number or range. This information is the one updated by the test. The test also removes flaws from the known_flaws file if they don't appear in the Bandit output.
+    Default: `wazuh`
+- `--reference`: the repository branch. 
 
-#### Parameters
+    Default: `master`
```

Comment:
I think it could be a good idea to use a table to show parameters values. For example something like this
| Parameter             | Description                                           | Default Value     |
|-----------------------|-------------------------------------------------------|-------------------|
| `--repo`              | The repository to test.                              | `wazuh`           |
| `--reference`         | The repository branch.                                | `master`          |
| `--check_directories` | The directories to check (comma-separated).           | `framework/,api/,wodles/` |
| `--exclude_directories`| The directories to exclude (comma-separated).        | `test/,tests/`    |
| `--confidence`        | Minimum confidence level for Bandit scan.            | `MEDIUM`           |
| `--severity`          | Minimum severity level for Bandit scan.              | `LOW`             |

IMHO it is faster to find the desired information